### ISSUE TITLE
%appdata%\Temp\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey

### ISSUE THREAD

**Issue by Oxymoron290** [chocolatey/choco#1661]
Too much chocolate. JDK can't handle it. Install fails with `2018-10-14 11:38:17,142 8648 [ERROR] - jdk8 not installed. An error occurred during installation:
 The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.`

```
2018-10-14 11:38:16,050 8648 [DEBUG] - XmlConfiguration is now operational
2018-10-14 11:38:16,094 8648 [INFO ] - ============================================================
2018-10-14 11:38:16,276 8648 [DEBUG] - Attempting to create directory "C:\Users\Oxymoron290\AppData\Local\Temp\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey".
2018-10-14 11:38:16,341 8648 [INFO ] - Chocolatey v0.10.11
2018-10-14 11:38:16,349 8648 [DEBUG] - Chocolatey is running on Windows v 10.0.17134.0
2018-10-14 11:38:16,351 8648 [DEBUG] - Attempting to delete file "C:/ProgramData/chocolatey/choco.exe.old".
2018-10-14 11:38:16,352 8648 [DEBUG] - Attempting to delete file "C:\ProgramData\chocolatey\choco.exe.old".
2018-10-14 11:38:16,361 8648 [DEBUG] - Command line: "C:\ProgramData\chocolatey\choco.exe" install jdk8
2018-10-14 11:38:16,362 8648 [DEBUG] - Received arguments: install jdk8
2018-10-14 11:38:16,409 8648 [DEBUG] - RemovePendingPackagesTask is now ready and waiting for PreRunMessage.
2018-10-14 11:38:16,416 8648 [DEBUG] - Sending message 'PreRunMessage' out if there are subscribers...
2018-10-14 11:38:16,423 8648 [DEBUG] - [Pending] Removing all pending packages that should not be considered installed...
2018-10-14 11:38:16,479 8648 [DEBUG] - The source 'https://chocolatey.org/api/v2/' evaluated to a 'normal' source type
2018-10-14 11:38:16,481 8648 [DEBUG] - 
NOTE: Hiding sensitive configuration data! Please double and triple 
 check to be sure no sensitive data is shown, especially if copying 
 output to a gist for review.
2018-10-14 11:38:16,489 8648 [DEBUG] - Configuration: CommandName='install'|
CacheLocation='C:\Users\Oxymoron290\AppData\Local\Temp\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey\chocolatey'|

ContainsLegacyPackageInstalls='True'|
CommandExecutionTimeoutSeconds='2700'|WebRequestTimeoutSeconds='30'|
Sources='https://chocolatey.org/api/v2/'|SourceType='normal'|
Debug='False'|Verbose='False'|Trace='False'|Force='False'|Noop='False'|
HelpRequested='False'|UnsuccessfulParsing='False'|RegularOutput='True'|
QuietOutput='False'|PromptForConfirmation='True'|AcceptLicense='False'|
AllowUnofficialBuild='False'|Input='jdk8'|AllVersions='False'|
SkipPackageInstallProvider='False'|PackageNames='jdk8'|
Prerelease='False'|ForceX86='False'|OverrideArguments='False'|
NotSilent='False'|ApplyPackageParametersToDependencies='False'|
ApplyInstallArgumentsToDependencies='False'|IgnoreDependencies='False'|
AllowMultipleVersions='False'|AllowDowngrade='False'|
ForceDependencies='False'|Information.PlatformType='Windows'|
Information.PlatformVersion='10.0.17134.0'|
Information.PlatformName='Windows 10'|
Information.ChocolateyVersion='0.10.11.0'|
Information.ChocolateyProductVersion='0.10.11'|
Information.FullName='choco, Version=0.10.11.0, Culture=neutral, PublicKeyToken=79d02ea9cad655eb'|

Information.Is64BitOperatingSystem='True'|
Information.Is64BitProcess='True'|Information.IsInteractive='True'|
Information.UserName='Oxymoron290'|
Information.UserDomainName='DESKTOP-OV7N3RV'|
Information.IsUserAdministrator='True'|
Information.IsUserSystemAccount='False'|
Information.IsUserRemoteDesktop='False'|
Information.IsUserRemote='True'|
Information.IsProcessElevated='True'|
Information.IsLicensedVersion='False'|Information.LicenseType='Foss'|
Features.AutoUninstaller='True'|Features.ChecksumFiles='True'|
Features.AllowEmptyChecksums='False'|
Features.AllowEmptyChecksumsSecure='True'|
Features.FailOnAutoUninstaller='False'|
Features.FailOnStandardError='False'|Features.UsePowerShellHost='True'|
Features.LogEnvironmentValues='False'|Features.LogWithoutColor='False'|
Features.VirusCheck='False'|
Features.FailOnInvalidOrMissingLicense='False'|
Features.IgnoreInvalidOptionsSwitches='True'|
Features.UsePackageExitCodes='True'|
Features.UseFipsCompliantChecksums='False'|
Features.ShowNonElevatedWarnings='True'|
Features.ShowDownloadProgress='True'|
Features.StopOnFirstPackageFailure='False'|
Features.UseRememberedArgumentsForUpgrades='False'|
Features.IgnoreUnfoundPackagesOnUpgradeOutdated='False'|
Features.RemovePackageInformationOnUninstall='False'|
Features.ScriptsCheckLastExitCode='False'|
ListCommand.LocalOnly='False'|
ListCommand.IdOnly='False'|ListCommand.IncludeRegistryPrograms='False'|
ListCommand.PageSize='25'|ListCommand.Exact='False'|
ListCommand.ByIdOnly='False'|ListCommand.ByTagOnly='False'|
ListCommand.IdStartsWith='False'|ListCommand.OrderByPopularity='False'|
ListCommand.ApprovedOnly='False'|
ListCommand.DownloadCacheAvailable='False'|
ListCommand.NotBroken='False'|
ListCommand.IncludeVersionOverrides='False'|
UpgradeCommand.FailOnUnfound='False'|
UpgradeCommand.FailOnNotInstalled='False'|
UpgradeCommand.NotifyOnlyAvailableUpgrades='False'|
UpgradeCommand.ExcludePrerelease='False'|
NewCommand.AutomaticPackage='False'|
NewCommand.UseOriginalTemplate='False'|SourceCommand.Command='unknown'|
SourceCommand.Priority='0'|SourceCommand.BypassProxy='False'|
SourceCommand.AllowSelfService='False'|
SourceCommand.VisibleToAdminsOnly='False'|
FeatureCommand.Command='unknown'|ConfigCommand.Command='unknown'|
PinCommand.Command='unknown'|OutdatedCommand.IgnorePinned='False'|
Proxy.BypassOnLocal='True'|
2018-10-14 11:38:16,490 8648 [DEBUG] - _ Chocolatey:ChocolateyInstallCommand - Normal Run Mode _
2018-10-14 11:38:16,495 8648 [INFO ] - Installing the following packages:
2018-10-14 11:38:16,498 8648 [INFO ] - jdk8
2018-10-14 11:38:16,499 8648 [INFO ] - By installing you accept licenses for the packages.
2018-10-14 11:38:17,098 8648 [INFO ] - [NuGet] Installing 'jdk8 8.0.181'.
2018-10-14 11:38:17,104 8648 [ERROR] - jdk8 not installed. An error occurred during installation:
 The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
2018-10-14 11:38:17,114 8648 [INFO ] - jdk8 package files install completed. Performing other installation steps.
2018-10-14 11:38:17,119 8648 [DEBUG] - No package information as package is null.
2018-10-14 11:38:17,135 8648 [DEBUG] - No package information to save as package is null.
2018-10-14 11:38:17,137 8648 [DEBUG] - Sending message 'HandlePackageResultCompletedMessage' out if there are subscribers...
2018-10-14 11:38:17,138 8648 [ERROR] - The install of jdk8 was NOT successful.
2018-10-14 11:38:17,142 8648 [ERROR] - jdk8 not installed. An error occurred during installation:
 The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
2018-10-14 11:38:17,155 8648 [WARN ] - 
Chocolatey installed 0/1 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
2018-10-14 11:38:17,157 8648 [INFO ] - 
2018-10-14 11:38:17,159 8648 [ERROR] - Failures
2018-10-14 11:38:17,161 8648 [ERROR] -  - jdk8 (exited 1) - jdk8 not installed. An error occurred during installation:
 The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.
2018-10-14 11:38:17,163 8648 [WARN ] - 
Enjoy using Chocolatey? Explore more amazing features to take your
experience to the next level at
 https://chocolatey.org/compare
2018-10-14 11:38:17,166 8648 [DEBUG] - Sending message 'PostRunMessage' out if there are subscribers...
2018-10-14 11:38:17,170 8648 [DEBUG] - Exiting with 1
```

**Comment by Oxymoron290** [comment]
p.s. all packages are failing with the same error. It isn't just the JDK. nodeJS also failed with the same error

**Comment by Oxymoron290** [comment]
https://github.com/chocolatey/choco/pull/1211

**Comment by DanJ210** [comment]
Is there any instruction on how to fix this? I have an issue present in my Visual Studio installer that just says there's a problem with Visual Studio Build Tools 2017 (2) with "Couldn't download Microsoft.VisualCpp.Premium.Tools.HostX64.TargetX64.Resources" and I don't know anything about it.

**Comment by heaths** [comment]
@DanJ210 you can either install Build Tools outside of Chocolatey for now, or try upgrading choco itself since this was fixed:

```
choco upgrade --yes chocolatey
```

**Comment by ferventcoder** [comment]
I believe this is seen with Boxstarter, and I've seen a corresponding issue. This hasn't been an issue with Chocolatey itself since #1210 was fixed in 0.10.4 almost 2 years ago. 
Here is that issue - https://github.com/chocolatey/boxstarter/issues/241

If anyone doesn't confirm they are seeing this in regular choco.exe without boxstarter, we can reopen this issue.
